src/main/java/org/argos/file/manager/controller/FileManagerController.java

```diff
@@ -0,0 +1,61 @@
+package org.argos.file.manager.controller;
+
+import java.util.List;
+import java.util.Map;
+import org.argos.file.manager.service.S3FileService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+/**
+ * REST Controller for managing files in AWS S3.
+ */
+@RestController
+@RequestMapping("/api")
+public class FileManagerController {
+
+    private final S3FileService s3FileService;
+
+    /**
+     * Constructor-based dependency injection for S3FileService.
+     *
+     * @param s3FileService the S3 file service.
+     */
+    @Autowired
+    public FileManagerController(S3FileService s3FileService) {
+        this.s3FileService = s3FileService;
+    }
+
+    /**
+     * Uploads an entire directory to the S3 bucket and generates a new project ID.
+     *
+     * @param localDir the path to the local directory to upload.
+     * @return a map containing the generated project ID and uploaded file statuses.
+     */
+    @PostMapping("/upload")
+    public Map<String, Object> uploadDirectory(@RequestParam String localDir) {
+        return s3FileService.uploadDirectory(localDir);
+    }
+
+    /**
+     * Lists all files in the S3 bucket for a specific project.
+     *
+     * @param projectId the ID of the project.
+     * @return a list of file keys in the S3 bucket for the given project.
+     */
+    @GetMapping("/files")
```

Comment:
add CORS to your endpoints please to allow make petitions from the front end
 ```@CrossOrigin(origins = "http://localhost:5173")```

Reply:
![image](https://github.com/user-attachments/assets/61f74200-8682-41da-9e84-c1af07ef261b)
Solved :heavy_check_mark: